File: backend/src/main/kotlin/models/classes/medication/Diagnosis.kt
package org.example.models.classes.medication

import org.example.models.classes.persons.Doctor
import org.example.models.classes.persons.Patient
import java.util.Date

data class Diagnosis(
    val id: Long,
    val medication: Long,
    val disease: String,
    val diagnosedBy: Doctor,
    val diagnosedPatient: Patient,
    val diagnosedAt: Date
)

File: frontend/src/types/Diagnosis.ts
import type { PatientApi } from './Patient'
import type { DoctorApi } from './Doctor.tsx'
import type { DrugApi } from './Drug.tsx'

export type DoseApi = {
  id: number
  unit: string
  amount: number
  frequency: string
  frequencyAmount: number
}

export type MedicationApi = {
  id: number
  dose: DoseApi
  drug: DrugApi
  started: string
  ended: string | null
}

export type DiagnosisApi = {
  id: number
  disease: string
  medication: MedicationApi | null
  diagnosedBy: DoctorApi
  diagnosedPatient: PatientApi
  diagnosedAt: string
  diagnosedEnd: string | null
}

export type DiagnosesResponse = {
  diagnoses: DiagnosisApi[]
}

export type MedicationCreateRequest = {
  dose_id: number
  drug_id: number
  started: string | null
  ended: string | null
}

export type DiagnosisCreateRequest = {
  disease: string
  medication_id: number
  diagnosed_by: number
  diagnosed_patient: number
  diagnosed_at: string
}
